pkg/v1/tkg/tkgctl/helpers_test.go

```diff
@@ -211,3 +212,66 @@ var _ = Describe("Cluster Class - IP Family Validation related test cases: ", fu
 		})
 	})
 })
+
+var _ = Describe("Test cases for CheckIfInputFileIsClusterClassBased", func() {
+	Context("Test cases for CheckIfInputFileIsClusterClassBased", func() {
+		var (
+			configFile          string
+			configFileContent   string
+			isClusterClassBased bool
+			err                 error
+		)
+
+		JustBeforeEach(func() {
+			configFile, err = utils.CreateTempFile("", "")
+			Expect(err).To(BeNil())
+			err = utils.SaveFile(configFile, []byte(configFileContent))
+			Expect(err).To(BeNil())
+			isClusterClassBased, _, err = CheckIfInputFileIsClusterClassBased(configFile)
+		})
+		When("File contains cluster resource with clusterclass defined", func() {
+			BeforeEach(func() {
+				configFileContent = `apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: aws-workload-cluster1
+  namespace: default
+spec:
+  topology:
+    class: tkg-aws-default`
+			})
+			It("should return true and error should be nil", func() {
+				Expect(isClusterClassBased).To(BeTrue())
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+		When("File contains cluster resource without clusterclass defined", func() {
+			BeforeEach(func() {
+				configFileContent = `apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  name: aws-workload-cluster1
+  namespace: default
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks: []
+`
+			})
+			It("should return false with error", func() {
+				Expect(isClusterClassBased).To(BeFalse())
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(ContainSubstring(""))
+			})
+		})
+		When("File doesn't contain cluster resource", func() {
+			BeforeEach(func() {
+				configFileContent = ``
+			})
+			It("should return false with error", func() {
```

Comment:
Should the description of the test say "without error"?  That is what the check below seems to expect.